syntax = "proto3";

package order;

// Mensaje que representa una solicitud para crear un pedido
message OrderRequest {
    string product_name = 1;         // Nombre del producto
    float price = 2;                 // Precio del producto
    string payment_gateway = 3;      // Pasarela de pago
    string card_brand = 4;           // Marca de la tarjeta
    string bank = 5;                 // Banco utilizado
    string shipping_address_region = 6; // Región de la dirección de envío
    string shipping_address = 7;     // Dirección de envío
    string customer_email = 8;       // Correo electrónico del cliente
}

// Mensaje que representa la respuesta del servidor al crear un pedido
message OrderResponse {
    string message = 1;              // Mensaje de respuesta
}

// Servicio que define las operaciones relacionadas con los pedidos
service OrderService {
    rpc CreateOrder(OrderRequest) returns (OrderResponse); // Método para crear un pedido
}